examples/advanced/xgboost_secure/README.md

```diff
@@ -0,0 +1,118 @@
+
+To be able to run all the examples, please install the requirements first.
+
+```
+pip install -r requirements.txt
+```
+
+## Data Preparation
+### Download and Store Data
+To run the examples, we first download the dataset from this [link](https://www.kaggle.com/datasets/mlg-ulb/creditcardfraud), which is a single `.csv` file.
+By default, we assume the dataset is downloaded, uncompressed, and stored in `${PWD}/dataset/creditcard.csv`.
+
+> **_NOTE:_** If the dataset is downloaded in another place,
+> make sure to modify the corresponding `DATASET_PATH` inside `prepare_data.sh`.
+
+### Data Split
+To prepare data for further experiments, we perform the following steps:
+1. Split the dataset into training/validation and testing sets. 
+2. Split the training/validation set: 
+    * Into train and valid for baseline centralized training.
+    * Into train and valid for each client under horizontal setting. 
+    * Into train and valid for each client under vertical setting.
+
+Data splits used in this example can be generated with
+```
+bash prepare_data.sh
+```
+
+This will generate data splits for 3 clients under all experimental settings. In this example, we assume the Private Set Intersection (PSI) step has already been performed for vertical collaboration.
+See [vertical xgboost](https://github.com/NVIDIA/NVFlare/tree/main/examples/advanced/vertical_xgboost) for more details. With this assumption, the overlapping ratio between clients for vertical setting is 1.0, such that the training data amount is the same as baseline and horizontal experiments.
+
+> **_NOTE:_** The generated data files will be stored in the folder `/tmp/nvflare/xgb_dataset/`,
+> and will be used by jobs by specifying the path within `config_fed_client.json` 
+
+## Run Baseline Training
+First, we run the baseline centralized training with:
+```
+bash run_training_central.sh
+```
+
+## Run Federated Experiments with NVFlare
+Next, we run the federated XGBoost training without and with homomorphic encryption using NVFlare. 
+We run the NVFlare jobs using simulator with: 
+```
+bash run_training_fl.sh
+```
+The running time of each job depends mainly on the encryption workload. 
+
+Comparing the AUC results with centralized baseline, we have three observations:
+1. The performance of the model trained with homomorphic encryption is identical to its counterpart without encryption.
+2. Vertical federated learning (both secure and non-secure) have identical performance as the centralized baseline.
+3. Horizontal federated learning (both secure and non-secure) have performance slightly different from the centralized baseline. This is because under horizontal FL, the local histogram quantiles are based on the local data distribution, which may not be the same as the global distribution.
+
+The AUC of vertical learning (both secure and non-secure):
+
+[0]	eval-auc:0.90515	train-auc:0.92747
+[1]	eval-auc:0.90516	train-auc:0.92748
+[2]	eval-auc:0.90518	train-auc:0.92749
```

Comment:
maybe clarify what 0,1,2 refer to, are those the client sites?

Reply:
Ah right, those are number of rounds.

Each site getting the same numbers now.

Let me clarify this